### PR TITLE
For #23424 - Add context menu for contile top sites

### LINKED ISSUE #23424: Add context menu for Contile Top Sites
https://mozilla-hub.atlassian.net/browse/FNXV2-18723

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #23424 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemMenu.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    context.getString(R.string.rename_top_site)
+                ) {
+                    onItemTapped.invoke(Item.RenameTopSite)
+                }
+            } else {
+                null
+            },
+            if (isPinnedSite) {
+                SimpleBrowserMenuItem(
+                    if (isPinnedSite) {
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemMenu.kt:56]: was this duplicate if intentional?

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemMenu.kt:56]: will update my PR, we basically want to show this menu item for pinned and frecent sites (aka all sites that are not TopSite.Provided)

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemMenu.kt:56]: Resolved in latest changes. We check for `!isProvidedSite`

**Review by MozillaNoah — APPROVED** [review]
LGTM

**Pull request merged**